packages/liquidator/src/liquidator.js

```diff
@@ -251,51 +274,118 @@ class Liquidator {
       );
 
       // Simple version of inventory management: simulate the transaction and assume that if it fails, the caller didn't have enough collateral.
-      let liquidationResult, gasEstimation;
-      try {
-        [liquidationResult, gasEstimation] = await Promise.all([
-          liquidation.call({ from: this.account }),
-          liquidation.estimateGas({ from: this.account })
+      const notEnoughCollateral = await liquidation
```

Comment:
Thinking through worst case scenarios: is it possible that `liquidation.call()` could fail for other reasons besides the liquidator not having enough funds? For example, if it doesn't have enough allowance, or the `minCollateralPerToken`, `maxCollateralPerToken`, or `tokensToLiquidate` input vars are invalid?

Reply:
There are many reasons - but lets assume that for now. We can add those logic later on

Reply:
hm I don't think you can actually make this assumption, because false positives for `liquidation.call()` failing will result in wasted gas from oneinch txns, right? Isn't it easier to just calculate the amount of collateral needed to run the liquidation before hand, and then if `collateral.balanceOf <` this amount then you can do the one-inch?

Reply:
Collateral needed would be `tokensToLiquidate/tokensOutstandingInPosition * collateralInPosition`

Reply:
I'm basing the assumption off your comment - https://github.com/UMAprotocol/protocol/blame/c50aa4765b3a762f74c28ad9f0ae7da5bad1cac8/packages/liquidator/src/liquidator.js#L253

Reply:
Ah I see. That comment is pretty outdated at this point and incorrect. I think we can be more specific with the one-inch integration and just check if we need allowance explicitly. It's not a difficult check: `tokensToLiquidate/tokensOutstandingInPosition * collateralInPosition`

Reply:
Sure. But lets not keep shifting the goal post - this should have been brought up in https://github.com/UMAprotocol/protocol/issues/1804

Reply:
Great point, I've posted an updated comment here which should contain all the relevant info to make this catch more robust: https://github.com/UMAprotocol/protocol/issues/1804#issuecomment-679135307

Reply:
Even an invalid liquidation against a position that someone else liquidated first could cause this to error out. We've had this before when the bots state updates too slowly and it tries to liquidate a position after they withdrew their collateral. See this [tx](https://etherscan.io/tx/0x38ab85be13845648f275084342d6e19ae5ddbbad462805e1a88c3dcb0451179f) for example.

I think your current implementation would result in the failed TX _buying_ synthetics in some cases if the `.call` was to fail.